services/worker-manager/src/providers/google.js

```diff
@@ -208,63 +199,31 @@ class GoogleProvider extends Provider {
 
     // First check to see if the request is coming from the project this provider manages
     if (dat.project_id !== this.project) {
-      const error = new Error(`Invalid project ${dat.project_id} is not ${this.project}`);
-      error.project = dat.project_id;
-      error.validProject = this.project;
-      throw error;
+      throw new RegistrationError('Token project mismatch');
```

Comment:
I have a question: I'm not sure if this guards against some accidentals or against a malicious action..... if the latter, maybe we shouldn't be as specific in the response? For example, Github throws 404 for authentication/authorization errors.

Reply:
It is the latter.  That's what Brian was suggesting, too.  And it's what we do for bad authentication (401 with "Bad Mac").  I suppose it's pretty unlikely to see any of these errors when things are operating correctly. And I just realized that the most likely error a malicious actor would trigger is an exception from validateIdToken -- which this code handles as a 500.  So, I'll rewrite to always return the same error in any of those cases.